app/scripts/controllers/network/provider-api-tests/helpers.ts

```diff
@@ -156,7 +148,7 @@ function mockRpcCall({ nockScope, request, response, error, delay, times }) {
       }
     }
   }
-  const url = nockScope.basePath.includes('infura.io')
+  const url = (nockScope as any).basePath.includes('infura.io')
```

Comment:
Nit: Maybe at least worth a comment on why `any` is used here (the type is missing this property).

Totally fine with cleaning this up in a later PR (or after the merger), but a comment would help us keep track of the reason it was done in the first place

Reply:
Yeah. For the record it's because `basePath` isn't included in the TypeScript types for Nock even though it's a part of the public API. We could fix this in Nock though and that would allow us to fix it on the core side. I'll make a note of that.

Reply:
For the time being I've replaced this typecast with a `@ts-expect-error` line: [1494bda](https://github.com/MetaMask/metamask-extension/pull/18490/commits/1494bda32dc8e5520bc0b53d1deaa33f76472cbb). I'll try to make sure this gets included on the core side too.